test/gtest/uct/ib/test_devx.cc

```diff
@@ -49,3 +51,216 @@ UCS_TEST_P(test_devx, dbrec)
 }
 
 UCT_INSTANTIATE_IB_TEST_CASE(test_devx);
+
+
+class test_devx_umr_mkey : public test_devx {
+public:
+    void init() {
+        test_devx::init();
+    }
+
+    bool check_xgvmi() const {
+        return md()->flags & UCT_IB_MLX5_MD_FLAG_INDIRECT_XGVMI;
+    }
+
+    /*
+     * XGVMI capability advertised by device might be dropped on the initial
+     * operation. This function performs dummy XGVMI operation to detect
+     * whether XGVMI is actually operational
+     */
+    bool check_xgvmi_effective() const {
```

Comment:
can we unite check_xgvmi_effective and check_xgvmi to a single function?

Reply:
why do we care about testing "fake-gvmi" devices? IMO we should check if xgvmi is "really" supported (or not)

Reply:
Because those "fake-xgvmi" devices are real devices that we support. For instance, when I run my test on rock, I see that both mlx5_0 and mlx5_2 (MT4129 == ConnectX7?) are examples of those devices. I've no idea why they advertise XGVMI capability. But since we **can** run export mkey on those devices, I prefer to guarantee that it works as expected (= not using UMR approach).